lib/cn/kblock_builder.c

```diff
@@ -1188,9 +1208,9 @@ kbb_finish(struct kblock_builder *bld, struct blk_list *kblks, u64 seqno_min, u6
      */
     if (!kblock_is_empty(&bld->curr)) {
         struct wbb *pt = 0;
-        u64         kbsize = bld->rp->kblock_size;
-        u64         ptsize = (wbb_page_cnt_get(bld->ptree)) * PAGE_SIZE;
-        u64         kbused =
+        uint32_t    kbsize = bld->max_size;
```

Comment:
What's the intent behind using the max mblock size in this function vs determining the right mblock size for the age group of this kvset + DTYPE_KEY?

All kblocks in a kvset are allocated from the same media class as there's no media class allocation fallback support.